apis/datadoghq/v1alpha1/datadogagent_types.go

```diff
@@ -1405,7 +1405,6 @@ const (
 
 // DatadogAgent Deployment with Datadog Operator.
 // +kubebuilder:object:root=true
-// +kubebuilder:subresource:status
```

Comment:
Without deleting this line, `controller-gen` fails with below error. Since `v1alpha1` is getting deprecated in this release I don't removing this as a problem so didn't debug it any further
```
❯ bin/darwin-arm64/controller-gen "crd:preserveUnknownFields=false",crdVersions=v1 rbac:roleName=manager-role paths="./apis/..." output:crd:artifacts:config=config/crd/bases/v1
[repo_root]/apis/datadoghq/v1alpha1/datadogagent_types.go:1418:6: status subresource applied to version "v1alpha1" not in CRD
Error: not all generators ran successfully
```